ProcessOptimizer/plots.py

```diff
@@ -1555,6 +1556,228 @@ def plot_objective_1d(
+
+    * `size` [float, default=3]
+        Height (in inches) of each returned figure.
+
+    * `max_quality` [int, default=5] 
+        The maximal quality intended for the setup of Brownie Bee. Quality is
+        assumed to be measured on a scale from 0 to this number, and the y-axis
+        of each plot is scaled to reflect this. If the uncertainty reaches above
+        this number, the y-axis is expanded to accomodate this.
+
+    Returns
+    -------
+    * `plot_list`: [`Figures`]:
+        A list of individual matplotlib figure handles, one for each dimension
+        present in 'result' and a last one representing a histogram of samples
+        drawn at the expected minimum.
+    """
+   
+    space = result.space
+    # Check if we have any categorical dimensions, as this influences the plots
+    is_cat = [isinstance(dim, Categorical) for dim in space.dimensions]
+    # Check if we have any integer dimensions, as this influences the plots
+    is_int = [isinstance(dim, Integer) for dim in space.dimensions]
+    # Identify the location of the expected minimum, and its mean and std
+    x_eval, [res_mean, res_std] = expected_minimum(
+        result,
+        n_random_starts=20,
+        random_state=None,
+        return_std=True,
+    )
+        
+    rvs_transformed = space.transform(space.rvs(n_samples=n_samples))
+    _, minimum, _ = _map_categories(space, result.x_iters, x_eval)
+    
+    # Gather all data relevant for plotting
+    plots_data = []
+    # Starting point for our maximum visible quality
+    plot_max_q = max_quality
+    
+    for i in range(space.n_dims):
+        row = []
+        xi, yi, stddevs = dependence(
+            space,
+            result.models[-1],
+            i,
+            j=None,
+            sample_points=rvs_transformed,
+            n_points=n_points,
+            x_eval=x_eval,
+        )
+        row.append({"xi": xi, "yi": yi, "std": stddevs})
+        plots_data.append(row)
+        # Check if our quality interval goes above our desired max_quality view
+        if -np.min(yi-1.96*stddevs) > plot_max_q:
+            plot_max_q = -np.min(yi-1.96*stddevs)
+    
+    # Create the list to store figure handles
+    figure_list = []  
+    
+    # Expand maximal quality value if needed to show upper bound on uncertainty
+    max_quality = plot_max_q
+    
+    # Build all the plots in the figure
+    for n in range(space.n_dims):
+        # Prepare a figure    
+        fig, ax_ = plt.subplots(
+            figsize=(size, size),
+            dpi=200,
+        )
+        # Set the padding
+        fig.subplots_adjust(
+            left=0.12, right=0.93, bottom=0.2, top=0.95, hspace=0.0, wspace=0.0
+        )
+
+        # Get data to plot in this subplot
+        xi = plots_data[n][0]["xi"]
+        yi = plots_data[n][0]["yi"]
+        stddevs = plots_data[n][0]["std"]        
+        
+        # Set y-axis limits with a small buffer
+        ax_.set_ylim(0, max_quality*1.02)

```

Comment:
Do we always want 0 to be the worst quality shown? We probably don't want to plot even worse quality, but what if all values are between 3 and 5? Do we then "waste" half our "paper" by showing where nothing happens? That would mean showing fewer details than we could.

Reply:
We don't really want to remove the "bottom" anchor to make the axis fully floating, since this opens the door for extremely narrow axes if a model becomes very confident of being flat in a narrow region of Y values.